index.js

```diff
@@ -34,6 +34,13 @@ class Option {
     }
     this.description = description || '';
     this.defaultValue = undefined;
+    this.negateOption = false;
+  }
+
+  allowNegateOption(allowNegateOption) {
+    allowNegateOption = (allowNegateOption === undefined) || allowNegateOption;
+    this.negateOption = !this.negate && this.long && !this.required && allowNegateOption;
```

Comment:
I don't think the `required` test is appropriate. Commander allows `--no-bar` to be added when there is a `--bar <value>`.

Reply:
Seems you are right.